File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/settings/exceptions/SettingNotFoundException.java
package com.hgianastasio.biblioulbrav2.core.settings.exceptions;

/**
 * Created by heitor_12 on 06/06/17.
 */

public class SettingNotFoundException extends RuntimeException {
    private String key;

    public SettingNotFoundException(String key) {
        this.key = key;
    }

    @Override
    public String getMessage() {
        return "A configuração ".concat(key).concat(" não foi encontrada.");
    }
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/settings/exceptions/SettingNotFoundException.kt
package com.hgianastasio.biblioulbrav2.core.settings.exceptions

/**
 * Created by heitor_12 on 06/06/17.
 */
class SettingNotFoundException(private val key: String) : RuntimeException() {
    override val message: String
        get() = "A configuração $key não foi encontrada."

}